Doc/c-api/unicode.rst

```diff
@@ -307,6 +307,51 @@ These APIs can be used for fast direct character conversions:
    possible.  This function does not raise exceptions.
 
 
+.. c:function:: Py_ssize_t PyUCS4_ToLower(Py_UCS4 ch, Py_UCS4 *buffer, Py_ssize_t size)
+
+   Convert *ch* to lower case, store result in *buffer*, which should be
+   able to hold as many characters needed for *ch* to be lower cased
+   (e.g. a maximum of two character for Unicode 16.0), and
+   return the number of characters stored. If at some point a buffer overflow
+   is detected, an :exc:`ValueError` is raised and ``-1`` is returned.
```

Comment:
The word “maximum” makes it sound like you can't set *size* to more than 2.
Perhaps something like this? 
```suggestion
   able to hold as many characters needed for *ch* to be lower cased, and
   return the number of characters stored. If at some point a buffer overflow
   is detected, an :exc:`ValueError` is raised and ``-1`` is returned.

   In Unicode 16.0, any character can be lowercased into a buffer of *size* ``2``.
   See also :c:macro:`PyUCS4_CASE_CONVERSION_BUFFER_SIZE`.
```

Please also update `/home/encukou/dev/cpython/Tools/unicode/makeunicodedata.py` -- around line 40 there's a list of docs pages that refer to Python's current Unicode version.